src/Sylius/Bundle/CoreBundle/CatalogPromotion/Announcer/CatalogPromotionArchivalAnnouncer.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\CatalogPromotion\Announcer;
+
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\ArchiveCatalogPromotion;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\DisableCatalogPromotion;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\RestoreCatalogPromotion;
+use Sylius\Bundle\CoreBundle\CatalogPromotion\Command\UpdateCatalogPromotionState;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class CatalogPromotionArchivalAnnouncer implements CatalogPromotionArchivalAnnouncerInterface
+{
+    public function __construct(private MessageBusInterface $commandBus)
+    {
+    }
+
+    public function dispatchCatalogPromotionArchival(CatalogPromotionInterface $catalogPromotion): void
+    {
+        $this->commandBus->dispatch(new UpdateCatalogPromotionState($catalogPromotion->getCode()));
+
+        if ($catalogPromotion->isEnabled()) {
+            $this->commandBus->dispatch(new DisableCatalogPromotion($catalogPromotion->getCode()));
+        }
+
+        $this->commandBus->dispatch(new ArchiveCatalogPromotion($catalogPromotion->getCode()));
+    }
+
+    public function dispatchCatalogPromotionRestoral(CatalogPromotionInterface $catalogPromotion): void
+    {
+        $this->commandBus->dispatch(new RestoreCatalogPromotion($catalogPromotion->getCode()));
```

Comment:
Should it `enable` catalog promotion on restore as it's disabling it on archivation?

Reply:
No, in my opinion it shouldn't. From archived promotions perspective you don't know whether the promotion was enabled before archiving or not.